deploy/kiali/kiali_cr.yaml

```diff
@@ -268,6 +273,23 @@ spec:
 #    ---
 #    ingress_enabled: true
 #
+# The instance name of this Kiali installation. This instance name will be the prefix
+# prepended to the names of all Kiali resources created by the operator and will be used
+# to label those resources as belonging to this Kiali installation instance.
+# You cannot change this instance name after a Kiali CR is created. If you attempt
+# to change it, the operator will abort with an error. If you want to change it,
+# you must first delete the original Kiali CR and create a new one.
+# Note that this does not affect the name of the auto-generated signing key secret. If
+# you do not supply a signing key, the operator will create one for you in a secret,
+# but that secret will always be named "kiali-signing-key" and shared across all
+# Kiali instances in the same deployment namespace. If you want a different
+# signing key secret, you are free to create your own and tell the operator about it via
+# the login_token.signing_key setting. See the docs on that setting for more details.
+# Note also that if you are setting this value, you may also want to change
+# the installation_tag setting, but this is not required.
+#    ---
+#    instance_name: "kiali"
```

Comment:
What about using the name of the `Kiali` CR as the prefix instead of adding a new deployment field? It could work the same way where if your `Kiali` CR is named `kiali` then you don't get a prefix, everything is just called `kiali` but if your CR is named something like `admin` then the prefix would be `admin-kiali`. That's more similar to how some kube controllers name their managed resources like the `Deployment` controller and the pods associated with it. Might be less of a backwards compatible change depending on how users have named their `Kiali` CRs though.

Reply:
The Kiali server helm chart would still need a separate "prefix" value to set.

Reply:
A couple things:
1) right, we want to maintain consistency between how helm charts configure Kiali and how the operator does things - having a config setting allows for documentation to refer to one way to do this ("set deployment.instance_name") and it works either way. A person moving from helm to operator doesn't need to know about this "special" meaning of the Kiali CR name - all settings are in the spec. field just as they are in the helm values file.
2) Putting things in the CR like this PR does it (inside spec) means that setting automatically goes in the ConfigMap which is accessible to the Server - so the Server could get this setting if it needs to (and I suspect it may need to see this instance_name at some point in the future). And, again, this would allow for the Server to be installer-agnostic (it doesn't need to know it was installed via helm or operator - it just looks in the ConfigMap and gets the setting. Otherwise, it would have to know "Oh, I was installed by the Operator? I need to look up the Kiali CR (and it may not even have access to it or know where it is) and get its name. or... Oh, I was installed by helm? I need to look in the ConfigMap,). I suppose the server could just get its pod name and deduce what the instance_name is, but that's a pretty annoying "special case" if it wants to get a user configuration setting - when all the other ones are in the ConfigMap.

In short, we like to keep all user configuration settings in the Kiali CR spec field so they get transferred to the ConfigMap thus maintaining consistency across all user config and also making it the config available to the Server.